traits/observation/_generated_parser.py

```diff
@@ -1665,443 +2332,834 @@ def from_ParseTable(cls, parse_table):
{'@': 26})}, 10: {17: (0, 0), 18: (0, 7), 8: (0, 29), 9: (0, 13), 16: (1, {'@': 23}), 11: (1, {'@': 23})}, 11: {10: (1, {'@': 30}), 11: (1, {'@': 30})}, 12: {17: (0, 0), 18: (0, 7), 8: (0, 29), 9: (0, 13), 16: (1, {'@': 24}), 11: (1, {'@': 24})}, 13: {5: (1, {'@': 15}), 3: (1, {'@': 15}), 2: (1, {'@': 15}), 1: (1, {'@': 15}), 14: (1, {'@': 15})}, 14: {16: (1, {'@': 18}), 9: (1, {'@': 18}), 11: (1, {'@': 18}), 8: (1, {'@': 18}), 10: (1, {'@': 18})}, 15: {8: (1, {'@': 22}), 16: (1, {'@': 22}), 9: (1, {'@': 22}), 11: (1, {'@': 22})}, 16: {1: (0, 23), 2: (0, 25), 19: (0, 20), 7: (0, 1), 13: (0, 31), 0: (0, 17), 15: (0, 11), 20: (0, 21), 3: (0, 18), 14: (0, 5), 4: (0, 14), 21: (0, 33), 5: (0, 2), 6: (0, 4)}, 17: {16: (1, {'@': 16}), 9: (1, {'@': 16}), 11: (1, {'@': 16}), 8: (1, {'@': 16}), 10: (1, {'@': 16})}, 18: {8: (1, {'@': 11}), 16: (1, {'@': 11}), 9: (1, {'@': 11}), 11: (1, {'@': 11}), 10: (1, {'@': 11})}, 19: {8: (1, {'@': 21}), 16: (1, {'@': 21}), 9: (1, {'@': 21}), 11: (1, {'@': 21})}, 20: {11: (0, 26), 10: (1, {'@': 33})}, 21: {}, 22: {8: (1, {'@': 21}), 9: (1, {'@': 21}), 10: (1, {'@': 27}), 11: (1, {'@': 27})}, 23: {2: (0, 28)}, 24: {15: (0, 34), 0: (0, 17), 1: (0, 23), 2: (0, 25), 3: (0, 18), 14: (0, 5), 4: (0, 14), 5: (0, 2), 6: (0, 4), 7: (0, 22)}, 25: {8: (1, {'@': 10}), 16: (1, {'@': 10}), 9: (1, {'@': 10}), 11: (1, {'@': 10}), 10: (1, {'@': 10})}, 26: {1: (0, 23), 2: (0, 25), 5: (0, 2), 7: (0, 1), 13: (0, 31), 0: (0, 17), 15: (0, 11), 3: (0, 18), 14: (0, 5), 4: (0, 14), 21: (0, 30), 6: (0, 4)}, 27: {16: (0, 6), 11: (0, 32)}, 28: {8: (1, {'@': 12}), 16: (1, {'@': 12}), 9: (1, {'@': 12}), 11: (1, {'@': 12}), 10: (1, {'@': 12})}, 29: {5: (1, {'@': 14}), 3: (1, {'@': 14}), 2: (1, {'@': 14}), 1: (1, {'@': 14}), 14: (1, {'@': 14})}, 30: {10: (1, {'@': 31}), 11: (1, {'@': 31})}, 31: {17: (0, 24), 18: (0, 3), 8: (0, 29), 9: (0, 13)}, 32: {0: (0, 17), 1: (0, 23), 2: (0, 25), 3: (0, 18), 13: (0, 10), 4: (0, 14), 5: (0, 2), 6: (0, 4), 7: (0, 15)}, 33: {10: (1, {'@': 32}), 11: (1, {'@': 32})}, 34: {10: (1, {'@': 28}), 11: (1, {'@': 28})}, 35: {8: (1, {'@': 20}), 9: (1, {'@': 20}), 10: (1, {'@': 25}), 11: (1, {'@': 25})}}, 'start_states': {'start': 16}, 'end_states': {'start': 21}}, '__type__': 'ParsingFrontend'}, 'rules': [{'@': 10}, {'@': 11}, {'@': 12}, {'@': 13}, {'@': 14}, {'@': 15}, {'@': 16}, {'@': 17}, {'@': 18}, {'@': 19}, {'@': 20}, {'@': 21}, {'@': 22}, {'@': 23}, {'@': 24}, {'@': 25}, {'@': 26}, {'@': 27}, {'@': 28}, {'@': 29}, {'@': 30}, {'@': 31}, {'@': 32}, {'@': 33}], 'options': {'debug': False, 'keep_all_tokens': False, 'tree_class': None, 'cache': False, 'postlex': None, 'parser': 'lalr', 'lexer': 'contextual', 'transformer': None, 'start': ['start'], 'priority': 'normal', 'ambiguity': 'auto', 'regex': False, 'propagate_positions': False, 'lexer_callbacks': {}, 'maybe_placeholders': False, 'edit_terminals': None, 'g_regex_flags': 0, 'use_bytes': False, 'import_paths': [], 'source_path': None, '_plugins': {}}, '__type__': 'Lark'}
```

Comment:
I naively assumed that I could run `python -m lark.tools.standalone traits/observation/_dsl_grammar.lark --out traits/observation/_generated_parser.py` on my personal machine and check that there are no changes in the generated parser code.

That doesn't seem to be the case. This dictionary gets modified when I regenerate the parser code. The change just moved around the key/value pairs in the `parser` `tokens` dictionary.

Reply:
Hmm, interesting; that'll be due to hash randomization, I guess. https://github.com/lark-parser/lark/issues/595 looks related.

Possibly we should be generating using a fixed `PYTHONHASHSEED`.

Reply:
> It's not clear to me what the source of the non-determinism is.

From a brief look at the Lark source, I suspect that the source of non-determinism is memory addresses of objects: the code is using `id`s of objects as set elements and/or dictionary keys. (Which then affects iteration order, etc.) There _is_ a use of `random.randint` in `earley_forest.py`, but I don't think we exercise that code path.

I'll stop trying to find a way to make this deterministic and merge as-is.